actionpack/lib/action_dispatch/middleware/cookies.rb

```diff
@@ -458,6 +460,15 @@ def make_set_cookie_header(header)
         def write_cookie?(cookie)
           request.ssl? || !cookie[:secure] || always_write_cookie
         end
+
+        def set_expires_attribute(options)
+          expires_at, expires_in = options[:expires_at], options[:expires_in]
+          if expires_at
+            options[:expires] = expires_at
+          elsif expires_in
+            options[:expires] = expires_in.from_now
+          end
+        end
```

Comment:
This makes sense to me. The expiration set for a cookie can match the expiration for the underlying Verified/Encrypted message. This would definitely mitigate the "valid forever" cookie problem as the messages will now have their own expiration builtin. :+1: